core/templates/domain/story/story-validation.service.ts

```diff
@@ -52,7 +55,11 @@ export class StoryValidationService {
     // prerequisite skills required by the destination nodes 'unlocked' by
     // visiting a particular node by the time that node is finished.
     while (nodesQueue.length > 0) {
-      var currentNodeIndex = storyContents.getNodeIndex(nodesQueue.shift());
+      // 'shift()' returns 'undefined' only when NodesQueue array is empty,
+      // the loop terminates before that condition is reached.
+      // non-null assertion (!) is used to make typescript typing happy.
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      var currentNodeIndex = storyContents.getNodeIndex(nodesQueue.shift()!);
```

Comment:
I've tackled this here https://github.com/oppia/oppia/pull/13394/files#r672509934. I think `getNodeIndex` should return -1 for all invalid values making undefined a valid case here. Can you make the same changes to that file instead ?

Reply:
done